internal/dag/httpproxy_processor.go

```diff
@@ -1221,6 +1231,9 @@ func (p *HTTPProxyProcessor) processHTTPProxyTCPProxy(validCond *contour_api_v1.
 				TCPHealthCheckPolicy: healthPolicy,
 				SNI:                  s.ExternalName,
```

Comment:
Looks like setting SNI for non-ExternalName services is still a TODO. It seems to me that we may need to add an API field for it, since it's not possible to always infer what should be set.

Reply:
Yes I left it to another PR as it is not 100% clear how to to infer that. 

Currently for normal `httpproxy` we have the [logic](https://github.com/projectcontour/contour/blob/61b6fae59b3262058b666337cc1fb8d12b0fa1a6/internal/dag/httpproxy_processor.go#L1601-L1620) based on `requestHeadersPolicy` with key [`Host`](https://projectcontour.io/docs/1.27/config/api/#projectcontour.io/v1.HeadersPolicy). For extension service use [`extensionservice.spec.validation.subjectName(s)`](https://github.com/projectcontour/contour/blob/main/internal/dag/extension_processor.go#L170-L177).  In theory we could do that for all of the cases by default, but then I started to wonder if it is a good idea: what normal TLS clients do by default is just to set SNI to the hostname it is connecting to. That would be what I'd expect, but maybe this discussion is better to have on another PR :)

Reply:
Yeah, discussing fully in another issue/PR sounds good to me.

Reply:
xref #6098